bundles/org.openhab.binding.paradoxalarm/src/main/java/org/openhab/binding/paradoxalarm/internal/handlers/ParadoxIP150BridgeHandler.java

```diff
@@ -0,0 +1,276 @@
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Start initialize()...");
+        updateStatus(ThingStatus.UNKNOWN);
+        logger.info("Starting creation of communicator.");
+        config = getConfigAs(ParadoxIP150BridgeConfiguration.class);
+        updateStatus(ThingStatus.OFFLINE);
+
+        scheduler.execute(this::initializeCommunication);
+        logger.debug("Finished initialize().");
+    }
+
+    private void initializeCommunication() {
+        try {
+            communicator = initializeCommunicator();
+            if (communicator.isOnline()) {
+                logger.info("Communicator created successfully. Update Data cache...");
+                updateDataCache(true);
+
+                updateStatus(ThingStatus.ONLINE);
+            }
+        } catch (IOException | InterruptedException | ParadoxBindingException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Error initializing panel handler. Exception: " + e);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        cancelSchedule(refreshCacheUpdateSchedule);
+        communicator.close();
+        super.dispose();
+    }
+
+    private void scheduleRefresh() {
+        logger.debug("Scheduling cache update. Refresh interval: {}s. Starts after: {}s.", config.getRefresh(),
+                INITIAL_SCHEDULE_DELAY);
+        refreshCacheUpdateSchedule = scheduler.scheduleWithFixedDelay(this::updateDataCache, INITIAL_SCHEDULE_DELAY,
+                config.getRefresh(), TimeUnit.SECONDS);
+    }
+
+    private void updateDataCache() {
+        updateDataCache(false);
+    }
+
+    private void updateDataCache(boolean withEpromValues) {
+        try {
+            logger.debug("Refreshing memory map");
+            RawStructuredDataCache cache = RawStructuredDataCache.getInstance();
+
+            boolean isOnline = communicator.isOnline();
+            cache.setIsOnline(isOnline);
+
+            if (isOnline) {
+                communicator.refreshMemoryMap();
+
+                cache.setPanelInfoBytes(communicator.getPanelInfoBytes());
+                cache.setPartitionStateFlags(communicator.readPartitionFlags());
+                cache.setZoneStateFlags(communicator.readZoneStateFlags());
+
+                if (withEpromValues) {
+                    cache.setPartitionLabels(communicator.readPartitionLabels());
+                    cache.setZoneLabels(communicator.readZoneLabels());
+                }
+            }
+
+            announceUpdateToHandlers();
+
+            if (failedReadAttempts > 0) {
+                logger.info("Successfully refreshed memory map after {} failed attempts.", failedReadAttempts);
+                failedReadAttempts = 0;
+            }
+
+        } catch (IOException e) {
+            handleSocketReadError(e);
+        } catch (InterruptedException | ParadoxBindingException e) {

```

Comment:
If we receive this InterruptedException, the interrupted flag was cleared and we most likely need to reset the flag:

http://www.yegor256.com/2015/10/20/interrupted-exception.html
https://www.ibm.com/developerworks/library/j-jtp05236/